tests/api-diff/record_type_tests.t

```diff
@@ -92,3 +92,21 @@ Run api-watcher on the two cmi files
   +type y = int option
   
   [1]
+
+### Adding a type parameter to a record type declaration:
+
+  $ cat > add_type_parameter.mli << EOF
+  > type 'a student = {first_name: string; last_name: string; id: int option}
+  > EOF
+
+We generate the .cmi file
+
+  $ ocamlc add_type_parameter.mli << EOF
+
+Run api-watcher on the two cmi files
+
+  $ api-diff ref.cmi add_type_parameter.cmi
+  diff module Add_type_parameter:
+  +<unsupported change>
```

Comment:
This should not be reported as an unsupported change here, there's something fishy going on.

I'd expect this to be represented as an atomic type_decl diff and therefore dispayed as such.

The problem might come from somewhere else. Could you try to see if adding this test on top of main shows the same result?

It's probably best to take that test out of this PR anyway as it was either always there or was introduced in the record diffing PR.

Reply:
It's also an unsupported change on main, but also there like an output of an changed record type declaration.

Reply:
Can you extract this test out of this PR and make a new PR on top of main with just that test? we should fix this separately?

Reply:
my bad, you already did!